src/dbus2mqtt/dbus/dbus_util.py

```diff
@@ -22,3 +27,200 @@ def unwrap_dbus_objects(args):
+    
+    elif isinstance(value, float):
+        # Float values are fine as-is for D-Bus
+        return value
+    
+    elif isinstance(value, str):
+        # String values are fine as-is for D-Bus
+        return value
+    
+    else:
+        # For any other type, try to convert to string as fallback
+        logger.warning(f"Unknown type {type(value)} for D-Bus conversion, converting to string: {value}")
+        return str(value)
+
+def _get_dbus_signature(value: Any) -> str:
+    """
+    Get the appropriate D-Bus signature for a value.
+    
+    Args:
+        value: The value to get signature for
+        
+    Returns:
+        D-Bus type signature string
+    """
+    if isinstance(value, bool):
+        return 'b'  # boolean
+    elif isinstance(value, int):
+        UINT16_MIN = 0
+        UINT16_MAX = 0xFFFF
+        INT16_MIN = -0x7FFF - 1
+        INT16_MAX = 0x7FFF
+        UINT32_MIN = 0
+        UINT32_MAX = 0xFFFFFFFF
+        INT32_MIN = -0x7FFFFFFF - 1
+        INT32_MAX = 0x7FFFFFFF
+        UINT64_MIN = 0
+        UINT64_MAX = 18446744073709551615
+
+        if UINT16_MIN <= value <= UINT16_MAX:
+            return 'q'  # 16-bit unsigned int
+        elif INT16_MIN <= value <= INT16_MAX:
+            return 'n'  # 16-bit signed int
+        elif UINT32_MIN <= value <= UINT32_MAX:
+            return 'u'  # 32-bit unsigned int
+        elif INT32_MIN <= value <= INT32_MAX:
+            return 'i'  # 32-bit signed integer
+        elif UINT64_MIN <= value <= UINT64_MAX:
+            return 't'  # 64-bit unsigned integer
+        else:
+            return 'x'  # 64-bit signed integer
+    elif isinstance(value, float):
+        return 'd'  # double
+    elif isinstance(value, str):
+        return 's'  # string
+    elif isinstance(value, list):
+        if not value:
+            return 'as'  # assume array of strings for empty arrays
+        # Get signature of first element and assume homogeneous array
+        element_sig = _get_dbus_signature(value[0])
+        return f'a{element_sig}'  # array of elements
+    elif isinstance(value, dict):
+        return 'a{sv}'  # dictionary of string to variant
+    else:
+        return 's'  # fallback to string
+
+# Helper function specifically for ConnMan-style property setting
+def convert_connman_property_args(property_value: Any) -> Any:
+    """
+    Helper function specifically for ConnMan SetProperty calls.
+    
+    ConnMan SetProperty expects: SetProperty(variant property_value)
+    
+    Args:
+        property_value: Value to set (will be converted and wrapped in Variant)
+        
+    Returns:
+        List with Variant(converted_value)
+    """
+    converted_value = _convert_value_to_dbus(property_value)
+    signature = _get_dbus_signature(converted_value)
+    return Variant(signature, converted_value)
+
+# Alternative conversion that wraps all complex types in Variants
+def convert_mqtt_args_to_dbus_explicit_variants(args: List[Any]) -> List[Any]:
```

Comment:
I still find it strange that this isn't handled by the high-level APIs of dbus-fast. I was trying to avoid all this logic as much as possible :-). Please add unit tests so that it can be refactored at a later moment if needed.